cupyx/scipy/sparse/compressed.py

```diff
@@ -32,6 +33,227 @@ class _compressed_sparse_matrix(sparse_data._data_matrix):
         ''',
         'compress_getitem_complex')
 
+    _max_reduction_kern = core.RawKernel(r'''
+        extern "C" __global__
+        void max_reduction(double* data, int* x, int* y, int length,
+                           double* z) {
+            // Get the index of the block
```

Comment:
Please avoid comments that just describe the code.

Reply:
I intended for this to help inform how the indexing is handled since it gets more complicated to handle the varying number of elements in each row/column. My preference would be for keeping the comments since it will help with the overall understanding of the functions.

Reply:
In this case `Calculate the block length` is redundant since you are calculating a variable called `block_length`, but other comments such as `// Select initial value based on the block density` are needed 🙂